plugins/parsers/avro/parser.go

```diff
@@ -57,7 +57,11 @@ func (p *Parser) Init() error {
 		if p.Timestamp != "" {
 			return errors.New("if 'timestamp' field is specified, 'timestamp_format' must be as well")
 		}
-		if p.TimestampFormat != "unix" && p.TimestampFormat != "unix_us" && p.TimestampFormat != "unix_ms" && p.TimestampFormat != "unix_ns" {
+	} else {
+		switch p.TimestampFormat {
+		case "unix", "unix_ns", "unix_us", "unix_ms":
+			// Valid values
+		default:
 			return fmt.Errorf("invalid timestamp format '%v'", p.TimestampFormat)
 		}
 	}
```

Comment:
How about replacing line 56 to 67 (including) with
```golang
	switch p.TimestampFormat {
	case "":
		p.TimestampFormat = "unix"
	case "unix", "unix_ns", "unix_us", "unix_ms":
		// Valid values
	default:
		return fmt.Errorf("invalid timestamp format '%v'", p.TimestampFormat)
	}
```
as this is what the config has as defaults. You don't use the format if no `Timestamp` is given anyway...

Reply:
That does change the behavior of the test case where you don't specify the timestamp format, but probably for the better.